### ISSUE TITLE
Fix overflow of blocks in total contributions card

### ISSUE THREAD

**Issue by shivanigangadharan** [wikimedia/WikiContrib#131]
The monthly contributions blocks are overflowing the total contributions card from both sides. 
They need to be contained inside the card. One way to do this would be adding a scrollable feature to their outer div. 

![image](https://user-images.githubusercontent.com/30795425/76701872-3160da00-66eb-11ea-9bed-6e6eb6d3ef91.png)

**Comment by shivanigangadharan** [comment]
@rammanoj I would like to work on this issue and fix it using the scrollable feature.

**Comment by srish** [comment]
@shivanigangadharan thanks for reporting this issue! I think the best way to go about this would be to reduce the size of the green square boxes, remove the space between month blocks just like Github does.

On small screens, these blocks overflows even more: 

<img width="200" alt="Screen Shot 2020-03-16 at 3 57 38 PM" src="https://user-images.githubusercontent.com/288353/76806707-69295980-679f-11ea-9458-534a5621b95c.png">

So, it would be great if the responsiveness of this component could be addressed as well. It would nice to understand how Github handles this as well. On small screens, it seems like they reduce the number of months to show.

**Comment by shivanigangadharan** [comment]
> @shivanigangadharan thanks for reporting this issue! I think the best way to go about this would be to reduce the size of the green square boxes, remove the space between month blocks just like Github does.

> So, it would be great if the responsiveness of this component could be addressed as well. It would nice to understand how Github handles this as well. On small screens, it seems like they reduce the number of months to show.

Okay got it. Will work on this the way you mentioned.
Thanks! :)

**Comment by shivanigangadharan** [comment]
@srish I decreased the size of the blocks as you had mentioned, and it looks fine in the desktop view. But as the screen size decreases further it gets very cramped up. I observed GitHub too doesn't show all the blocks properly in smaller screens.
Now, I can either add a scrollbar (could be a vertical one for mobile view) or I could separate them into 2 rows each showing 6 months, using media query. I'd personally prefer the former one since it would be feasible for every screen size, but which one would you recommend?

**Comment by NdibeRaymond** [comment]
I think this issue is only observed in large laptop screens and tablets (I haven't tested for tablets so I base my judgement on the image @srish shared above). If so, I believe it'd be ok to look at how it was implemented in Android screens and use that to implement it for tablets. Horizontal scrolling will be bad.
Below is how it looks on my android:
![20200317-162845](https://user-images.githubusercontent.com/40905613/76872318-9263eb80-686c-11ea-9c2b-a23421ec5cdd.png)
![20200317-162839](https://user-images.githubusercontent.com/40905613/76872323-93951880-686c-11ea-8ad8-2fca5838463d.png)
So your main focus I think should be the large computer screens (and maybe tablets, you can test for that one)

**Comment by shivanigangadharan** [comment]
Ahh. I think @srish might have first opened it in laptop mode and probably just decreased the screen size to see in mobile view. Whereas @NdibeRaymond might have viewed it directly in the mobile view. 
Seems that the blocks are placed according to the **first** screen size they're opened in. Once you try to resize the screen, they won't adjust to the new screen width but stay put according to the previous one. Although that is resolved when we refresh the screen in the new mode. 

To sum it up, I'll fix it for the laptop and tablet mode, and mobile view shouldn't have any problem if opened directly. 👍

**Comment by srish** [comment]
@shivanigangadharan Yes, you are right! I opened it in laptop mode and then reduced the screen size. The block appears nicely on my phone, like in @NdibeRaymond's screenshot. 
So, yes, your judgment is right there to fix the issue for laptop & tablet mode. Do not add scrolling if possible; it won't look good.  If you can, it would be nice to investigate if there is another way to place blocks and not according to the first screen they were opened in.

**Comment by shivanigangadharan** [comment]
> @shivanigangadharan Yes, you are right! I opened it in laptop mode and then reduced the screen size. The block appears nicely on my phone, like in @NdibeRaymond's screenshot.
> So, yes, your judgment is right there to fix the issue for laptop & tablet mode. Do not add scrolling if possible; it won't look good. If you can, it would be nice to investigate if there is another way to place blocks and not according to the first screen they were opened in.

Fixed it for the laptop view, the tablet view was working fine previously as well. I avoided the scrolling and used media query to change the width. I am yet to fix the auto-resize of blocks but hope this much looks good for now. Will update on that too asap. Thanks for all your suggestions @srish and @NdibeRaymond 😄

**Comment by rammanoj** [comment]
@shivanigangadharan Are you really able to reproduce the above issue. I don't think there will be a issue in any device except the large screens (maybe). I have written the code in such a way that the page refreshes according to the screen size. If you are on laptop and trying to test it by decreasing the size of browser. Can you refresh the page when you go ot the desired measurements and see ?

If you do not refresh the page after changing the measurement of the browser (drastically). You will definitely see this issue. So, refresh the page when you change the size and see if the issue is still reproducible ?

**Comment by shivanigangadharan** [comment]
> @shivanigangadharan Are you really able to reproduce the above issue. I don't think there will be a issue in any device except the large screens (maybe). I have written the code in such a way that the page refreshes according to the screen size. If you are on laptop and trying to test it by decreasing the size of browser. Can you refresh the page when you go ot the desired measurements ?
> 
> If you do not refresh the page after changing the measurement of the browser (drastically). You will definitely see this issue. So, refresh the page when you change the size and see if the issue is still reproducible ?

You're absolutely right like I mentioned above the blocks are perfect when we refresh the page after resizing. So if a user directly views it from their mobile, there won't be this issue.
Other than that, yes, the blocks overflowed in large screen (laptop) so I fixed that in PR #162 
Let me know if there's anything left to work in this issue @srish @rammanoj Thanks! :)
